src/c/mod.rs

```diff
@@ -34,6 +34,29 @@ use crate::constants;
 use crate::types::{Account, Provider, Subscription};
 use crate::utils::watch_tx;
 
+fn client_clauses_validator(
+    client: *mut ToriiClient,
+    clauses: *const EntityKeysClause,
+    clauses_len: usize,
+) -> Result<bool> {
+    if client.is_null() {
+        return Result::Err(Error {
+            message: CString::new("Invalid client pointer").unwrap().into_raw(),
+        });
+    }
+    if clauses.is_null() {
```

Comment:
We should be able to have empty clauses

Reply:
Will implement a validator to return an empty `Vec` if clauses is null

Reply:
I tried going for an approach like this:
```rust
fn clause_validator(
    clauses: *const EntityKeysClause,
    clauses_len: usize
) -> Vec<torii_grpc::types::EntityKeysClause> {
    if clauses.is_null() || clauses_len == 0 {
        return Vec::new();
    } else {
        let clauses = unsafe { std::slice::from_raw_parts(clauses, clauses_len) };
        return clauses.iter().map(|c| c.into()).collect::<Vec<_>>();
    }
}
```
And then calling it inside the functions:
```rust
let clauses = clause_validator(clauses, clauses_len);
```

But I am encountering the crash anyways, any idea if I am missing something?

Reply:
mhh thats funky, maybe the torii client is null? since we dont check it in that clause_validator

Reply:
I was also checking for the client in another function (the original validator without the clauses check). But still crashed. If I put back the original validator it complained about the null clauses pointer like before. 

So I guess that is the issue in all cases, I tried attaching a debugger to the DLL but couldn't get much more information :/

Do you remember how you generated the empty slice before? I could not find it in previous commits

Reply:
Oh wait mhh maybe now it's crashing because Unity is giving it a dangling ptr instead of a null one? Did you pull the latest changes from the unity repo? Maybe try going back 1 PR behind and checkign if it works. 

For the empty slice, I'm pretty sure I was doing it like you did. Just passing an empty vec if our clauses ptr was null / len was 0